src/main/java/org/zalando/jackson/datatype/money/MonetaryAmountSerializer.java

```diff
@@ -54,7 +56,7 @@ public void serialize(final MonetaryAmount value, final JsonGenerator generator,
 
     @Nullable
     private String format(final MonetaryAmount value, final SerializerProvider provider) {
-        final Locale locale = provider.getConfig().getLocale();
+        final Locale locale = this.localeProvider.getCurrentLocale(provider);
```

Comment:
I just realized that we could pass this functionality to the `MonetaryAmountFormatFactory`. By default it will get the locale from the config (current behaviour). One could provide a wrapper around that factory that ignores the parameter and replaces it with something else:

```java
final class LocaleAwareMonetaryAmountFormatFactory implements MonetaryAmountFormatFactory {

    private final MonetaryAmountFormatFactory delegate;
    private final Supplier<Locale> currentLocale;

    LocaleAwareMonetaryAmountFormatFactory(final MonetaryAmountFormatFactory delegate, final Supplier<Locale> currentLocale) {
        this.delegate = delegate;
        this.currentLocale = currentLocale;
    }

    @Override
    public MonetaryAmountFormat create(final Locale locale) {
        return delegate.create(currentLocale.get());
    }
    
}
```

Reply:
@ptahchiev What's your opinion regarding this? ^

Reply:
@whiskeysierra TBH I don't like it - it will do the job but is kind-of ugly. You are completely ignoring the `locale` parameter. So if I were to inject the `MonetaryAmountFormatFactory` as a spring bean:
````
    @Resource(name = MonetaryAmountFormatFactoryImpl.NAME)
    private MonetaryAmountFormatFactory monetaryAmountFormatFactory;
````
and then use it:
````
 final MonetaryAmountFormat monetaryAmountFormat = monetaryAmountFormatFactory.create(Locale.US);
````
it could happen the result is formatted as GERMAN, even though I pass Locale.US as parameter. This will certainly confuse a lot of people.

Reply:
It looks ugly because of `MonetaryAmountFormatFactory` interface design which expects locale.
To make it better we need to change interface and move locale resolution inside of implementation.

One way is to change interface to have no arguments and another is to pass either `SerializationConfig` or `SerializerProvider` itself as an argument.